### PR TITLE
feat: Icon component using SVGR

### LINKED ISSUE #992: New Component: Icon
> :warning: **Note**: Creating this issue as a new "component" because it is listed by uswds as a component, _however_ there are many decisions to make around adding support internally and externally for the uswds icons. I'm going to compile some notes regarding my thoughts here, but before beginning work on this, I would suggest distilling via some discussion. The most important step in beginning this work will be to import uswds 2.10.0+ and see exactly how the icons are exposed to us, and how we can best wrap them. I don't _know_ that an Icon component is the way to go with this, but it seems like it could be good for us to use internally, and also expose for external use.

# [Icons](https://designsystem.digital.gov/components/icons/)

## USWDS Version

Icons have existed in USWDS previously to 2.10.0, but 2.10.0 added better default support and a broader selection of icons.

Not also that the icons have changed from fontawesome (also see not about docs in Other Considerations):
> **Active icons are found in** dist/img/usa-icons. The icons in this directory will be packaged into an SVG sprite. The complete set of available icons are located in two directories: dist/img/material-icons and dist/img/uswds-icons. Copy icons into a usa-icons directory within your project images directory and rebuild the SVG sprite using uswds-gulp.

## Description
Icons help communicate meaning, actions, status, or feedback.

## Requirements (proposed)
Implementation will largely depend on how the icons are externally exposed, and what importing them looks like. Below is an example thought

**Prop Interface**

```JSX
Props = {
  name: string // See bonus note below about this
  size?: 3 | 4 | 5 | 6 | 7 | 8 | 9 // Default `undefined` (scale to font size). Side note: Typescript ranges please https://github.com/Microsoft/TypeScript/issues/15480 
} & JSX.IntrinsicElements['svg']
```

## Example Implementation (proposed)

The `name` prop should be used to determine the icon path. Icon names match the filename in the [uswds source code](https://github.com/uswds/uswds/tree/develop/src/img/usa-icons). By inserting `name` into the import url for the icon, we can allow the user to select an icon without needing to know anything about the path or internal implementation, so long as they use the correct name from the list of [uswds icons](https://designsystem.digital.gov/components/icons/#all-icons). Just be sure that a name that does not match an existing icon errors acceptably.

```JSX
<Icon name="accessibility_new" size={6} />
```

> ### Bonus: 
> As mentioned above, while `name` is a `string` it would be nice to also have type safety (and IDE autocomplete) where only strings matching actual icon names are allowed. 
>
> If uswds icons in 2.10.0 are exported statically in a format that we can accomplish this with, I think that would be worth pursuing.
> 
> I recommend discussion and using personal judgement here.

**More Details**

- `<svg>`root element
- Default class `usa-icon`
- 

## Other Considerations

- ~~There are several places in the code where we use uswds icons. GovBanner uses a lock icon for example. As a part of this effort, it would be a good idea to evaluate replacing our icons internally with the icon component (especially wherever we may still be using font-awesome).~~
- ~~In our [styles and assets documentation](https://github.com/trussworks/react-uswds/blob/main/docs/styles_and_assets.md) we're recommending using fontawesome per uswds' recommendation (which no longer appears to be there, in favor of their internal list of icons). Our documentation surrounding Icons should be updated alongside this PR~~
- ~~USWDS examples set some intrinsic props to defaults (e.g. `focusable` and `role`). We should set the same defaults, but allow override if the corresponding prop is passed to the `Icon` component~~

Updated some of these other considerations to be a part of their own Issue: https://github.com/trussworks/react-uswds/issues/1303

### PR DESCRIPTION
**Pull request by suzubara**
# Summary

## Related Issues or PRs

This PR implements the USWDS Icon component using an alternative approach, converting each icon to inline SVGs that can be rendered as React components instead of a single `sprite.svg` file with the `<use>` element. I worked off the existing PR (https://github.com/trussworks/react-uswds/pull/1241/files) and used most of the existing `Icon` component code in an HOC that wraps each React SVG component with the supported props (ie, `size`) and adds the USWDS classes.

Closes #992 

## How To Test

* View the Icons in the deployed Storybook for this branch and verify they match the icons listed at [USWDS](https://designsystem.digital.gov/components/icon/). Note that the thumbs down icon changed in the [2.11.2 release](https://github.com/uswds/uswds/releases/tag/v2.11.2), so that asset will appear differently until we've updated to that version of USWDS.
* Check out the branch locally and start the Example app, and view the Icons page. It should display all of the new icons.
* If you have another app that uses this library, try installing this branch and verify you can import and render icon components in your app.

### Screenshots (optional)

image

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      role,
+      ...iconProps,
+    }
+
+    return <Component {...finalProps} />
+  }
+}
+
+export const AccessibilityNew: React.ComponentType<IconProps> =
+  makeUSWDSIcon(AccessibilityNewSvg)
```
Comment (ahobson) [inline src/components/Icon/Icon.tsx:46]: And so the plan would be to create one of these for each one of the icons from uswds, we at one point we had in the `icons.ts` file? Or did you want to try to automate the generation of this somehow?

**Review by ahobson — COMMENTED** [review]
Fabulous work!

File: example/src/pages/Icons.tsx [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   <section>
     <h1>Icons</h1>
-
     <p className="usa-intro">
-      We recommend using icons from <a href="https://github.com/FortAwesome/react-fontawesome">react-fontawesome</a> for projects that need icons.
+      We recommend using icons from{' '}
+      <a href="https://github.com/FortAwesome/react-fontawesome">
+        react-fontawesome
+      </a>{' '}
+      for projects that need icons.
```
Comment (brandonlenz) [inline example/src/pages/Icons.tsx:15]: I assume we want to update this to remove references to fontawesome and keep things exclusive to uswds icons?

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (suzubara) [reply src/components/Icon/Icon.tsx:46]: as much as I would like to automate this, it's taking a whole new effort to figure out and I'm inclined to just do the cheap & easy thing for now (knowing new icons won't be added unless the USWDS version changes). I will create a followup issue for automating this, possibly using a webpack plugin or script??

Comment (ahobson) [reply src/components/Icon/Icon.tsx:46]: I'm totally on board with a manual process!

File: src/components/Icon/Icons.ts [hunk 3]
```diff
@@ -0,0 +1,722 @@
+import AccessibilityNewSvg from 'uswds/dist/img/usa-icons/accessibility_new.svg?react'
```
Comment (suzubara) [inline src/components/Icon/Icons.ts:1]: Note the query param `?react` at the end of each import. That is what's telling webpack to load these files using `svgr`, and all other SVGs are still loaded using the asset loader (so SVGs imported as images or in CSS should not break)

Comment (ahobson) [reply src/components/Icon/Icons.ts:1]: That's neat! Total bikeshed, but would using `?svgr` maybe make it more obvious about what is going on? Feel free to ignore this comment :)

**Review by ahobson — DISMISSED** [review]
This is amazing. Thank you

File: src/components/Icon/Icons.ts (hunk 3 above)
Comment (suzubara) [reply src/components/Icon/Icons.ts:1]: OH I like that idea! tbh I got this from following an example elsewhere and at first didn't understand the use of `?react`. I will change it to `?svgr` as I think that is much clearer.

**Review by brandonlenz — DISMISSED** [review]
Seconding what Drew said. WOW! 

Happo Diff looked good too!

**Review by brandonlenz — APPROVED** [review]
Changes in d87838740e0d85aa53ad9166ec9524a3fbcfc0ed look good. Re-approved.

(Storybook deployment still looks great as well 🥳)

**Pull request merged**